### PR TITLE
Adding functional tests for plugins.

### PR DESCRIPTION
**Pull request by mdobson**

### CODE REVIEW

**Comment by mdobson** [comment]
WIP WIP WIP WIP

**Comment by mdobson** [comment]
Feel free to review. It's certainly a start.

File: test/spike-arrest-test.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    plugin.onrequest.apply(null, [req, res, onrequest_cb]);
+  });
+
+  it('will reject a requests that come in too fast', (done) => {
+    var onrequest_cb = () => {
+    };
+    
+    var onrequest_cb_second = (req, res, next) => {
+      assert.ok(req instanceof Error);
+      assert.equal(req.message, 'SpikeArrest engaged');
```
Comment (F1ERRO) [inline test/spike-arrest-test.js]: should this be res.message?

Comment (F1ERRO) [inline test/spike-arrest-test.js]: should this be `res.message`?

File: test/quota-test.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+const assert = require('assert');
+
+var exampleConfig = { 
+  EdgeMicroTestProduct: {
+    allow: '3',
+    interval: '1',
+    timeUnit: 'minute',
+    bufferSize: 10000,
+    uri: 'https://edgemicroservices-us-east-1.apigee.net/edgemicro/quotas/organization/ws-poc3/environment/test',
+    key: 'b7b69b6bd79e778782af9a653ae9c4d91e65440d8acbc7e969437a8a8dc13612',
```
Comment (F1ERRO) [inline test/quota-test.js]: not sure we want these hard coded in the tests, for security reasons.  What do you think?

File: test/cache-test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const cache = require('../cache/index');
```
Comment (F1ERRO) [inline test/cache-test.js]: should we get rid of this now that we are deleting the cache plugin?

File: quota-memory/index.js [hunk 4]
```diff
@@ -5,7 +5,9 @@ var Quota = require('volos-quota-memory');
 var debug = require('debug')('gateway:quota');
 
 module.exports.init = function(config, logger, stats) {
-
+  console.log('CONFIG START');
```
Comment (F1ERRO) [inline quota-memory/index.js]: can we delete this?

File: test/cache-test.js (hunk 3 above)
Comment (mdobson) [inline test/cache-test.js]: Yeah

File: quota-memory/index.js (hunk 4 above)
Comment (mdobson) [inline quota-memory/index.js]: Yeah

File: test/spike-arrest-test.js (hunk 1 above)
Comment (mdobson) [inline test/spike-arrest-test.js]: Nope. This plugin actually does things in a weird way, and passes an instance of Error from the plugin as the first argument.

**Comment by mdobson** [comment]
Updated per review comments.

File: test/quota-test.js (hunk 2 above)
Comment (F1ERRO) [inline test/quota-test.js]: lets make these env variables.  Other than that I think this is good to go.

**Comment by mdobson** [comment]
Added env variables for tests.

**Pull request merged**